src/librustc_mir/interpret/memory.rs

```diff
@@ -415,7 +415,8 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         id: AllocId,
         is_write: bool,
     ) -> InterpResult<'tcx, Cow<'tcx, Allocation<M::PointerTag, M::AllocExtra>>> {
-        let alloc = tcx.alloc_map.lock().get(id);
+        let alloc =
+            tcx.alloc_map.lock().get(M::resolve_thread_local_allocation_id(memory_extra, id));
```

Comment:
Hm... ok now I understand Ralf's comment more. The function is not used as I expected it would be. I'd expect that accessing a thread local static would allocate one thread local on the current thread. This site is accessing a pointer to a thread local, which is a different thing.

Basically I'd expect logic in the `GlobalAlloc::Static` branch below, not here.

Reply:
Does my answer to Ralf explain my reasoning and do you see how to improve it?